### PR TITLE
Add arm64 to build matrix and update cibuildwheel

### LINKED ISSUE #138: M1 compatibility
I noticed you are using cibuildwheel; it should be relatively easy to provide wheels for apple silicon. Are there any special reasons you are not doing so today?

### PR DESCRIPTION
**Pull request by Korijn**
Closes #138

* This upgrades CI to use the latest version of cibuildwheel and adds apple silicon wheels to the build configuration.
* Apple silicon wheels can only be built for python >=3.8.
* This newer version of cibuildwheel also happens to build musllinux wheels, so that's a free bonus, I guess!
* Split out the publish to pypi job

### CODE REVIEW

**Comment by rougier** [comment]
Waiting for build before merging.

**Comment by almarklein** [comment]
Side question: to what extend can we expect FreeType to be available on Linux systems? If it's always there, then we don't have to bother with the Linux-aarch64 builds (and others).

**Comment by Korijn** [comment]
> Side question: to what extend can we expect FreeType to be available on Linux systems? If it's always there, then we don't have to bother with the Linux-aarch64 builds (and others).

As far as I know freetype is not listed in any of the "manylinux" policies, so at least _they_ don't seem to think it's widely available.

**Comment by Korijn** [comment]
I pushed two small changes that make the builds a tiny bit faster.

**Comment by Korijn** [comment]
I think this can be merged!

File: .github/workflows/ci.yml
Comment (anthrotype) [inline .github/workflows/ci.yml:33]: the reason it was built on a lower python minor version was to make sure that the tests pass on the minimum supported version. Arguably 3.6 is too low nowadays (EOL soon at end of year), how about setting it to 3.7?

File: .github/workflows/ci.yml
Comment (anthrotype) [inline .github/workflows/ci.yml:50]: the whole building of sdist and publishing of wheels could be moved to a separate "deploy" job that is conditioned on the "build" job passing and is only triggered on tagged commits (so you don't have to repeat the `if:` condition every time).

**Review by anthrotype — APPROVED** [review]
LGTM with minor comments

File: .github/workflows/ci.yml
Comment (Korijn) [reply .github/workflows/ci.yml:33]: I set it back to 3.6 or 3.7, whatever you like, but arm64 needs 3.8 at a minimum, macos arm64 wheels aren't supported below that.

File: .github/workflows/ci.yml
Comment (Korijn) [reply .github/workflows/ci.yml:50]: Sure, I also prefer that but I didn't want to refactor your CI pipeline without consent :) I guess this is the green light to do that then.

File: .github/workflows/ci.yml
Comment (Korijn) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/ci.yml
Comment (Korijn) [reply .github/workflows/ci.yml]: Done.

**Comment by Korijn** [comment]
Ready for another review round.

**Comment by rougier** [comment]
Waiting for output before merging (I will need some ping or I'll forget to merge...)

**Comment by Korijn** [comment]
Passed!

**Comment by Korijn** [comment]
@rougier I think you can merge now :)

**Comment by Korijn** [comment]
One more ping @rougier to get this merged! :)

**Comment by rougier** [comment]
Merged!

**Pull request merged**
